packages/affine/shared/src/adapters/type.ts

```diff
@@ -17,16 +17,17 @@ export const isBlockSnapshotNode = (node: unknown): node is BlockSnapshot =>
 
 export type AdapterContext<
   ONode extends object,
-  TNode extends object,
-  TConverter extends DeltaASTConverter,
+  TNode extends object = never,
+  TConverter extends DeltaASTConverter = DeltaASTConverter,
 > = {
   walker: ASTWalker<ONode, TNode>;
   walkerContext: ASTWalkerContext<TNode>;
   configs: Map<string, string>;
   job: Job;
-  deltaConverter: TConverter;
+  deltaConverter?: TConverter;
   assets?: AssetsManager;
   updateAssetIds?: (assetsId: string) => void;
+  updateTextBuffer?: (text: string) => void;
```

Comment:
I feel that we can expose the whole text buffer to user instead of a single function as an access point.

Reply:
Got it, I'll make the changes to the code.